packages/medusa-plugin-meilisearch/src/loaders/index.js

```diff
@@ -1,5 +1,54 @@
+import { transformProduct } from "../utils/transform-products"
```

Comment:
I think all of the logic in here should be moved to the core so that on load the products will be indexed no matter what. The responsibility of the indexing should then be taken care of by the installed `searchService`. Flow should be something like this:

1. in loader use `productService.list` to fetch batches of products
2. then call `searchService.addDocuments(index, productBatch)`
3. transformations etc will be taken care of by the search service
4. search service calls provider specific endpoints to index batch

It might be that it becomes a bit tricky with `addDocuments` having to do conditional transformations based on the `index`, it could make sense to extend the Search API in this case so that search plugins would have to implement a `transformProducts` method. This would allow us to add a step: 

1b. call `searchService.transformProducts(productPatch)` 

The return value of the `transformProducts` method could then be used directly when calling `addDocuments(index, transformedBatch)`.

Doing it this way would mean that the search plugin wouldn't have to be aware of the `INDEX_NS` variable which seems to be a path to mistakes. 

We should maybe remove settings from the core Search API for now and allow plugins to control these themselves until we can find a more common approach?